### DISCUSSION TITLE
Shirking echarts bundle size

### DISCUSSION THREAD

**Discussion by CuteReimu** [vuepress-theme-hope/vuepress-theme-hope#4420] [Show and tell]
### Answers from the author

You can use [Echarts Online builder](https://echarts.apache.org/en/builder.html) to build your own echarts bundle or use [a custom entry](https://echarts.apache.org/handbook/en/basics/import/#shrinking-bundle-size) to load components you need. Then set it as alias of `echarts`.

**Comment by CuteReimu** [comment]
据我所知，babel-plugin-import是支持Echarts的。

另外，在Echarts官网上也提供了在线定制，只下载需要的模块的`echarts.min.js`的方式。

不过我对包体压缩确实不太熟悉，还是希望有对此了解的大佬提供一些帮助。

**Comment by Mister-Hope** [comment]
我本地试了一下，echarts 默认是有压缩好的 min.js 的，这很快，如果改用生成临时文件的方式，webpack的启动时间会极大增加。另外我看了一下，不方便给用户提供定制选项，非高级用户根本没有头绪他使用的表格用了哪些布局与组件。

因此就不再推进了。如果有更好的意见欢迎提出。

**Comment by Mister-Hope** [comment]
### Answers from the author

You can use [Echarts Online builder](https://echarts.apache.org/en/builder.html) to build your own echarts bundle or use [a custom entry](https://echarts.apache.org/handbook/en/basics/import/#shrinking-bundle-size) to load components you need. Then set it as alias of `echarts`.
